library/src/scala/tasty/reflect/SymbolOps.scala

```diff
@@ -14,6 +14,9 @@ trait SymbolOps extends Core {
     /** Flags of this symbol */
     def flags(implicit ctx: Context): Flags
 
+    /** Does the symbol match the flags? */
+    def is(flags: Flags)(implicit ctx: Context): Boolean
```

Comment:
There is already an `is` on the flags. This is redundant and should be removed from here. If you really want it you can add this extension method in the utils.

Reply:
It seems that it makes more sense to remove `Flags.is` and keep it here, from the usability point of view.

Reply:
Import `util` in order to use `symbol.is(flags)` is not friendly for programmers.

Reply:
I think there is a better way to organize the tasty reflect internals such that macro authors can write `symbol.is(flags)` without an additional import. Currently, this seems impossible without redundancy in tasty reflect.

We addressed the same problem before while experimenting with macros. The idea is to separate compiler-facing interface and macro-author-facing interface. The former corresponds to `core`, they should be minimal & no duplicate, it's a contract with Scala compilers. The latter  corresponds to `util`, they aim for friendliness, it's a contract with macro authors. The separation also helps manage API evolution by segregating concerns.

Macro authors do not need to care about `core` (the abstract methods can be protected thus not accessible), the default interface for macro authors is `util`.

Reply:
Wouldn't this required duplication of the code of the `core` in the `util`?

Reply:
Just discussed with @nicolasstucki and he has some idea to do the refactoring, will address in a later PR.